### PR TITLE
[INFRA-1131] Run Ldap on Kubernetes

### PR DESCRIPTION
**Pull request by olblak**
To Do:
- [x] Ldap container build from repo [jenkins-infra/ldap](https://github.com/jenkins-infra/ldap)
- [x] Ldap Backup/Restore process
- [ ] Validate SSL certificate configuration (Letsencrypt? *) -> won't be done
- [x] Validate Firewall rules
- [x] Healthcheck should query ldap database
- [ ] Tests
- [x] Fixed public IP from repo [jenkins-infra/azure](https://github.com/jenkins-infra/azure) 

* We need a way to dynamically reload configuration if certificate changed (every 3 months)

### CODE REVIEW

**Comment by rtyler** [comment]
@olblak I'll take a look at this once it's no longer a WIP

File: dist/profile/templates/kubernetes/resources/ldap/persistentVolumeClaim-backup.yaml.erb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+  name: backup
+  namespace: ldap
+spec:
+  storageClassName: ""
```
Comment (rtyler) [inline dist/profile/templates/kubernetes/resources/ldap/persistentVolumeClaim-backup.yaml.erb]: Will you add a comment on the significance of this so somebody (like me!) doesn't accidentally remove it in the future

File: hieradata/env/production.yaml [hunk 2]
```diff
@@ -30,6 +38,7 @@ profile::kubernetes::params::clusters:
...
+      ldap_public_ip: 52.232.180.203
```
Comment (rtyler) [inline hieradata/env/production.yaml]: This looks like hard-coded, can you add a comment where this hard-coded value comes from? Presumably terraform?

**Review by rtyler — APPROVED** [review]

**Review by rtyler — CHANGES_REQUESTED** [review]
bok

**Comment by olblak** [comment]
@rtyler  You can merge it once approved

**Review by rtyler — APPROVED** [review]

**Pull request merged**